src/gl/implementation/gl-objects-programs.adb

```diff
@@ -150,26 +151,77 @@ package body GL.Objects.Programs is
         (Program_Int_Param (Subject, Enums.Tess_Gen_Vertex_Order));
    end Tess_Gen_Vertex_Order;
 
-   function Transform_Feedback_Buffer_Mode (Subject : Program)
+   function Transform_Feedback_Buffer_Mode (Object : Program)
                                             return Buffer_Mode is
       function To_Buffer_Mode is new Ada.Unchecked_Conversion (Int, Buffer_Mode);
    begin
       return To_Buffer_Mode
-        (Program_Int_Param (Subject, Enums.Transform_Feedback_Buffer_Mode));
+        (Program_Int_Param (Object, Enums.Transform_Feedback_Buffer_Mode));
```

Comment:
The usual way to do this is to define `Program_Buffer_Mode_Param` and import that directly with the differently typed return value. Please do that instead of using `Ada.Unchecked_Conversion`.

Reply:
Have done (hopefully correctly)